src/libsyntax/print/pprust.rs

```diff
@@ -703,6 +703,7 @@ pub trait PrintState<'a>: std::ops::Deref<Target=pp::Printer> + std::ops::DerefM
         &mut self,
         path: &ast::Path,
         has_bang: bool,
+        ident: Option<ast::Ident>,
         tts: TokenStream,
         delim: MacDelimiter,
         span: Span,
```

Comment:
Maybe we could take `ast::Mac` as an argument here? It looks like that would at least somewhat cut down on the arguments to this function -- path, delim, tts, and span are all in `ast::Mac`. But I'm not sure if it'd actually be an improvement :/

Reply:
With `macro` support it's even further from `ast::Mac` now.

Reply:
Yeah, I think it's probably not worth the attempt to do something yet -- definitely something for a future refactor I think.